frontend/src/entities/dataset/dataset.actions.tsx

```diff
@@ -58,15 +60,102 @@ function DatasetActions (props?: any) {
     );
 }
 
-export const DatasetBrowserActions = (state: Pick<DatasetBrowserState, 'selectedItems' | 'items' | 'datasetContext' | 'manageMode' | 'filteringText'>, setState: ReduxDispatch<DatasetBrowserAction> | ReactDispatch<DatasetBrowserAction>, updateDatasetContext: UpdateDatasetContextFunction): React.ReactNode => {
+export async function fileHandler (arrayOfFiles: File[], manageMode: DatasetBrowserManageMode | undefined, notificationService: any, {state, setState, updateDatasetContext, setDisableUpload,}: Pick<UploadButtonProperties, 'state' | 'setState' | 'updateDatasetContext' | 'setDisableUpload'>) {
+    const filesToUpload = arrayOfFiles.map((file: File): DatasetResourceObject => ({
+        bucket: '',
+        type: 'object',
+        key: `${state.datasetContext?.location || ''}${file.webkitRelativePath ? file.webkitRelativePath : file.name}`,
```

Comment:
I think we could clean this up a little by using `${file.webkitRelativePath || file.name}` right?